variables.tf

```
@@ -1,10 +1,21 @@
-variable "saml_provider_name" {
-  description = "Name of the SAML Identity Provider"
-  type        = string
-}
-
-variable "saml_provider_metadata" {
-  description = "XML metadata for the SAML Identity Provider"
-  default     = ""
-  type        = string
+variable "iam_identity_provider" {
+  description = "Inputs to create a SAML Provider or OpenID Provider"
+  type = object({
+    saml = optional(object({
+      metadata_document = string
+      provider_name     = string
+    }))
+    openid_connect = optional(object({
+      client_id_list  = list(string)
+      thumbprint_list = optional(list(string), [])
+      url             = string
+    }))
+  })
+  validation {
+    condition = (
+      (var.iam_identity_provider.saml != null && var.iam_identity_provider.openid_connect == null) ||
+      (var.iam_identity_provider.saml == null && var.iam_identity_provider.openid_connect != null)
+    )
```

Had you seen the helper functions, anytrue and alltrue? They're handy for bundling up collections of conditions like this and somewhat simplifying the syntax to be more readable...

```suggestion
    condition = anytrue([
      var.iam_identity_provider.saml != null && var.iam_identity_provider.openid_connect == null,
      var.iam_identity_provider.saml == null && var.iam_identity_provider.openid_connect != null,
    ])
```

---

Updated condition